lib/pos_mempool.go

```diff
@@ -244,6 +251,18 @@ func (mp *PosMempool) Init(
 	mp.mempoolBackupIntervalMillis = mempoolBackupIntervalMillis
 	mp.maxValidationViewConnects = maxValidationViewConnects
 	mp.transactionValidationRefreshIntervalMillis = transactionValidationRefreshIntervalMillis
+	mp.recentBlockTxnCache = lru.NewKVCache(10000) // TODO: parameterize?
```

Comment:
You can make the capacity for this cache as high as 100k, and it would only take up 3MB. Pretty easy to make it bullet proof